src/krate.rs

```diff
@@ -668,6 +685,10 @@ pub fn new(req: &mut Request) -> CargoResult<Response> {
                                      .unwrap_or(&[]);
     let keywords = keywords.iter().map(|k| k[..].to_string()).collect::<Vec<_>>();
 
+    let categories = new_crate.categories.as_ref().map(|s| &s[..])
+                                     .unwrap_or(&[]);
+    let categories = categories.iter().map(|k| k[..].to_string()).collect::<Vec<_>>();
```

Comment:
`k[..]` what. I think this can just be `categories.iter().cloned().collect::<Vec<_>>();` Is the fish even required here; is `categories` constrained elsewhere?